torch/optim/adam.py

```diff
@@ -104,6 +110,9 @@ def step(self, closure=None):
                 bias_correction2 = 1 - beta2 ** state['step']
                 step_size = group['lr'] * math.sqrt(bias_correction2) / bias_correction1
 
+                if group['weight_decay'] != 0 and not group['l2_reg']:
+                    p.data.add_(-group['weight_decay'], p.data)
```

Comment:
I see, thank you for the clarification. :-)

I find it weird to decouple weight decay from learning rate. In my understanding, their claims would remain valid by applying the learning rate to the new weight decay implementation as well, without having weight decay going rogue.

This can lead to so many confusing results, because if one changes the learning rate, the decay remains just as strong as it was before. People would need to beware of both and adjust both accordingly for different experiments.

Reply:
ok then closing this as resolved...

Reply:
yes so the schedulers in this pull request also schedule the weight decay by the same `\eta_t` as the learning rate... so hopeful you don't need to think about adjusting the weight decay yourself.